lib/generator.js

```diff
@@ -642,9 +642,36 @@ class Generator {
     } catch (e) {
       this.templateConfig = {};
     }
+    await this.loadDefaultValues();
     await this.validateTemplateConfig();
   }
 
+  /**
+   * Loads default values of parameters from template config. If value was already set as parameter it will not be
+   * overrided.
+   * @private
+   */
+  async loadDefaultValues() {
+    const parameters = this.templateConfig.parameters;
+    const defaultValues = [];
+    Object.keys(parameters || {}).forEach(key => {
+      if (parameters[key].default) {
+        defaultValues.push(key);
+      }
+    });
+
+    if (Array.isArray(defaultValues)) {
```

Comment:
because of `const defaultValues = [];` it is always array and always `true` here

Reply:
Ok, will fix, but who say this to author of these lines https://github.com/asyncapi/generator/blob/230e6652e48f0215eb6834b53fc4eb86c6819677/lib/generator.js#L663-L668 ? 😃

Reply:
![](https://media.giphy.com/media/4pMX5rJ4PYAEM/200w_d.gif)

Reply:
fixed